react-forms/src/InputWithButton.tsx

```diff
@@ -0,0 +1,142 @@
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  0502/2024 - Preston Crawford
+//       Generated original version of source code.
+//
+// ******************************************************************************************************
+
+import * as React from 'react';
+import HelperMessage from './HelperMessage';
+import { CreateGuid, IsInteger, IsNumber } from '@gpa-gemstone/helper-functions'
+
+interface IProps<T> {
+    Record: T;
+    Field: keyof T;
+    Setter: (record: T) => void;
+    Valid: (field: keyof T) => boolean;
+    InputLabel?: string;
+    InputFeedback?: string;
+    InputDisabled?: boolean;
+    InputType?: 'number' | 'text' | 'password' | 'email' | 'color' | 'integer';
+    InputHelp?: string | JSX.Element;
+    InputStyle?: React.CSSProperties;
+    InputAllowNull?: boolean;
+    InputSize?: 'small' | 'large',
+    InputDefaultValue?: number,
+    OnBtnClick: (evt: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+    BtnLabel: string;
+    BtnClass?: string;
+    BtnDisabled?: boolean;
+    BtnStyle?: React.CSSProperties;
+}
+
+
+function InputWithButton<T>(props: IProps<T>) {
+    const internal = React.useRef<boolean>(false);
+    const [guid, setGuid] = React.useState<string>("");
+    const [showHelp, setShowHelp] = React.useState<boolean>(false);
+    const [heldVal, setHeldVal] = React.useState<string>(''); // Need to buffer tha value because parseFloat will throw away trailing decimals or zeros
+
+    React.useEffect(() => {
+        setGuid(CreateGuid());
+    }, []);
+
+    React.useEffect(() => {
+        if (!internal.current) {
+            setHeldVal(props.Record[props.Field] == null ? '' : (props.Record[props.Field] as any).toString());
+        }
+        internal.current = false;
+    }, [props.Record[props.Field]]);
+
+    function onBlur() {
+        const allowNull = props.InputAllowNull === undefined ? false : props.InputAllowNull;
+        if (!allowNull && (props.InputType === 'number' || props.InputType === 'integer') && heldVal === '') {
+            internal.current = false;
+            props.Setter({ ...props.Record, [props.Field]: props.InputDefaultValue ?? 0 });
```

Comment:
are we assuming it is a number? Why is the default `0` if not supplied?

Reply:
per our discussion `InputDefaultValue` is only used when `(props.InputType === 'number' || props.InputType === 'integer')` and `!allowNull` so a number is ok